Content.Server/StationEvents/Events/PowerGridCheck.cs

```diff
@@ -34,26 +44,50 @@ public override void Announce()
 
         public override void Startup()
         {
-            var entityManager = IoCManager.Resolve<IEntityManager>();
-
-            foreach (var component in entityManager.EntityQuery<ApcPowerReceiverComponent>(true))
+            foreach (var component in _entityManager.EntityQuery<ApcPowerReceiverComponent>(true))
             {
-                component.PowerDisabled = true;
                 _powered.Add(component.Owner);
             }
 
+            _numberPerSecond = Math.Max(1, (int)(_powered.Count / SecondsUntilOff)); // Number of APCs to turn off every second. At least one.
+
             base.Startup();
         }
 
-        public override void Shutdown()
+        public override void Update(float frameTime)
         {
-            var entMan = IoCManager.Resolve<IEntityManager>();
+            base.Update(frameTime);
+            _frameTimeAccumulator += frameTime;
+
+            var updates = 0;
+            if (_frameTimeAccumulator > UpdateRate)
+            {
+                updates = (int) (_frameTimeAccumulator / UpdateRate);
+                _frameTimeAccumulator -= UpdateRate * updates;
+            }
 
-            foreach (var entity in _powered)
+            for (var i = 0; i < updates; i++)
+            {
+                if (_powered.Count == 0)
+                    break;
+
+                var selected = _random.PickAndTake(_powered);
+                if (_entityManager.Deleted(selected)) continue;
+                if (_entityManager.TryGetComponent<ApcPowerReceiverComponent>(selected, out var powerReceiverComponent))
+                {
+                    powerReceiverComponent.PowerDisabled = true;
+                }
+                _unpowered.Add(selected);
```

Comment:
It was already like this before, but maybe make it only add to `_unpowered` if it was actually previously enabled.

That way the event wouldn't just randomly re-enable devices that were disabled by some other means.